src/base/logger.cpp

```diff
@@ -39,7 +39,7 @@ namespace
     QVector<T> loadFromBuffer(const boost::circular_buffer_space_optimized<T> &src, const int offset = 0)
     {
         QVector<T> ret;
-        ret.reserve(src.size() - offset);
+        ret.reserve(static_cast<typename QVector<T>::size_type>(src.size()) - offset);
```

Comment:
maybe use `static_cast<decltype(ret)::size_type>` to avoid specifying the container?
Also applies to other files.

Reply:
>maybe use `static_cast<decltype(ret)::size_type>` to avoid specifying the container?
Also applies to other files.

Done.